conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureBodySerDe.java

```diff
@@ -135,17 +136,44 @@ public void serialize(T value, HttpServerExchange exchange) throws IOException {
     private static final class EncodingDeserializerRegistry<T> implements Deserializer<T> {
 
         private final List<EncodingDeserializerContainer<T>> encodings;
+        private final boolean optionalType;
+        private final TypeMarker<T> marker;
 
         EncodingDeserializerRegistry(List<Encoding> encodings, TypeMarker<T> token) {
             this.encodings = encodings.stream()
                     .map(encoding -> new EncodingDeserializerContainer<>(encoding, token))
                     .collect(ImmutableList.toImmutableList());
+            this.optionalType = TypeMarkers.isOptional(token);
+            this.marker = token;
         }
 
         @Override
         public T deserialize(HttpServerExchange exchange) throws IOException {
+            if (optionalType && maybeEmptyBody(exchange)) {
```

Comment:
is it not possible to unit-test this type of stuff?

Reply:
I added two tests for this in `UndertowServiceEteTest`, though they go through the system end-to-end.

Reply:
yeah I saw those, but I find them notoriously hard to read and maintain because the cause/effect connection isn't so clear.

Reply:
Ack, I'll start by adding documentation to the implementation and tests. The unit tests for serialization have been difficult to keep in a state that represents real use. I worry that they make it more difficult to change implementation details, if they break too frequently we'll become desensitized to changing them to make them pass and fail to catch changes that cause bugs.

Reply:
OK, then the other alternative is to add these test cases to the conjure-verifier suite of tests.